src/core/interrupts.c

```diff
@@ -12,6 +12,7 @@
 
 BITMAP_ALLOC(hyp_interrupt_bitmap, MAX_INTERRUPTS);
 BITMAP_ALLOC(global_interrupt_bitmap, MAX_INTERRUPTS);
+spinlock_t irq_reserve_lock;
```

Comment:
This lock is never initialized.

```suggestion
spinlock_t irq_reserve_lock = SPINLOCK_INITVAL;
```

Reply:
That's a good point. Addressed in https://github.com/bao-project/bao-hypervisor/pull/97/commits/1a898b49d4bac726ca9a99e70feeb4affeb32ef0. Thank you for your feedback!